src/vs/languages/html/common/htmlWorker.ts

```diff
@@ -136,9 +136,9 @@ export class HTMLWorker {
 			return callback(false, model);
 		}
 
-		let modeAtPosition = modelAtPosition.getMode();
+		let modeAtPositionMode = modelAtPosition.getMode();
 
-		return callback(modeAtPosition.getId() !== this._modeId, modelAtPosition);
+		return callback(modeAtPositionMode.getId() !== this._modeId, modelAtPosition);
 	}
```

Comment:
The rename prevents a wrong overwrite of the modeAtPosition which then ends up with wrong calls in the getRangesToPosition.
getRangeToPosition in Mode html for selection grow/shrink doesn’t work because the function does not return anything if not embedded. I don’t have the in depth knowledge of the code to return the necessary implementation for the case that html is not embedded.

Von: Johannes Rieken [mailto:notifications@github.com]
Gesendet: Dienstag, 22. März 2016 12:13
An: Microsoft/vscode <vscode@noreply.github.com>
Cc: Stefan Schneider <Stefan.Schneider@connectiv.de>
Betreff: Re: [vscode] Bugfix: Prevent overwrite of modeAtPosition (#4505)


In src/vs/languages/html/common/htmlWorker.ts<https://github.com/Microsoft/vscode/pull/4505#discussion_r56969239>:

>

> -            return callback(modeAtPosition.getId() !== this._modeId, modelAtPosition);

> +            return callback(modeAtPositionMode.getId() !== this._modeId, modelAtPosition);

>       }

This is just a variable rename. Is there more to come?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/vscode/pull/4505/files/8b77f3c26f5924f683246d56ef8cb5f39e4c375b#r56969239>